### PR TITLE
extract product_dimensions from ingadhoc/odoo-addons

### PR DESCRIPTION
**Pull request by lepistone**
- [x] git extraction
- [x] PR to delete from the old repo
- [x] make flake8 pass
- [x] remove some text in spanish
- [x] check/delete comments
- [x] create a README
- [x] add the OCA as a second author

Discussed with the authors on ingadhoc/odoo-addons#90

### CODE REVIEW

**Comment by jjscarafia** [comment]
:+1:

**Comment by nhomar** [comment]
Hello @lepistone 

I understand it is not the objective of this pR explicitally.

But can you add only 1 test case with the creation of one product with this data on it and without this data to ensure it will not be broken ever.

Falling appart of that, :+1:

File: product_dimensions/product.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+# directory
+##############################################################################
+
+
+from openerp.osv import osv
+from openerp.osv import fields
+
+
+class Product(osv.osv):
+    _name = 'product.product'
```
Comment (pedrobaeza) [inline product_dimensions/product.py]: Remove this line

File: product_dimensions/product.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in module root
+# directory
+##############################################################################
+
+
+from openerp.osv import osv
```
Comment (pedrobaeza) [inline product_dimensions/product.py]: Compact these 2 lines in 1

File: product_dimensions/product.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+from openerp.osv import fields
+
+
+class Product(osv.osv):
```
Comment (pedrobaeza) [inline product_dimensions/product.py]: Use orm.Model instead, or better, use it new API :wink:

File: product_dimensions/product.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+                volume = length_m * heigth_m * width_m
+        v['volume'] = volume
+        return {'value': v}
+
+    def get_measure_in_meters(self, measure, dimensional_uom):
+        if not dimensional_uom:
+            return None
+
+        measure = float(measure)
+        if dimensional_uom.name == 'm':
```
Comment (pedrobaeza) [inline product_dimensions/product.py]: Why are you hard-coding this? Use the conversion method in product.uom model.

**Comment by pedrobaeza** [comment]
Please rename the module to match singular convention rule to _product_dimension_.

**Comment by lepistone** [comment]
I addressed all points, including adding tests, removing hardcoded UOM logic, new API, module rename.

File: product_dimension/README.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Product Dimension
+=================
+Add dimensions (length, width and height) products. Find the volume
```
Comment (pedrobaeza) [inline product_dimension/README.md]: s/) products/) to products

File: product_dimension/README.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Product Dimension
```
Comment (pedrobaeza) [inline product_dimension/README.md]: The title can still be "Product Dimensions" if you find it more useful

File: product_dimension/product.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    @multi
+    def onchange_calculate_volume(self, length, heigth, width,
+                                  dimensional_uom_id):
+        return {'value':
+                {'volume': self.compute_volume(length, heigth, width,
+                                               dimensional_uom_id)}
+                }
+
+    def convert_to_meters(self, measure, dimensional_uom):
+        UOM = self.env['product.uom']
```
Comment (pedrobaeza) [inline product_dimension/product.py]: We talked about finally not using caps words on model references.

File: product_dimension/product.py [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+        uom_meters = UOM.search([('name', '=', 'm')])
+        return UOM._compute_qty_obj(from_unit=dimensional_uom,
```
Comment (pedrobaeza) [inline product_dimension/product.py]: You can optimize this checking first if the UoM is different

Comment (lepistone) [inline product_dimension/product.py]: I prefer not not add a special case here.

File: product_dimension/product.py (hunk 7 above)
Comment (lepistone) [inline product_dimension/product.py]: But https://www.odoo.com/documentation/8.0/reference/guidelines.html says `ResPartner = self.env['res.partner']` and they started using that.

Comment (pedrobaeza) [inline product_dimension/product.py]: But check on https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md in section `Differences With Odoo Guidelines` that we have differed on this following this debate: https://github.com/OCA/maintainer-tools/pull/82

Comment (lepistone) [inline product_dimension/product.py]: OK!

**Comment by pedrobaeza** [comment]
:+1:

File: product_dimension/README.md [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+* https://www.odoo.com/forum/help-1
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/{project_repo}/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
+`here <https://github.com/OCA/{project_repo}/issues/new?body=module:%20{module_name}%0Aversion:%20{version}%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (laetitia-gangloff) [inline product_dimension/README.md]: You should replace value in {}

**Comment by laetitia-gangloff** [comment]
a little remark else :+1:

**Comment by lepistone** [comment]
@laetitia-gangloff done, thanks

**Comment by gurneyalex** [comment]
it would be nice to also port the onchange to the new api, as porting them later may break extensions to the module. I'd be :heart_eyes_cat: if you could do this @lepistone

**Pull request merged**